src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void validIndexCheck(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index: " + index);
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            head = node;
+            tail = head;
+        } else {
+            node = new Node<>(tail,value,null);
+            if (tail.prev == null) {
+                tail.prev = head;
+                head.next = node;
+            } else {
+                tail.next = node;
+                node.prev = tail;
+            }
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            validIndexCheck(index);
+        }
+        if (index == 0 && size != 0) {
+            Node node = new Node(null, value, head);
+            head.prev = node;
+            node.next = head;
+            head = node;
+            size++;
+        } else {
+            if (index == size || size == 0) {
+                add(value);
+            } else {
+                Node node = new Node(head, value, head.next);
+                for (int i = 1; i < index; i++) {
+                    node.prev = node.next;
+                    node.next = node.next.next;
+                }
+                node.next.prev = node;
+                node.prev.next = node;
+                size++;
+            }
+        }
```

Comment:
The logic in the add method by index is overly complicated and contains errors. For example, in line 64 you create a new node but you are not using it properly as you overwrite node.prev and node.next inside the loop. Consider simplifying this logic and maybe extracting some parts into private helper methods to avoid code duplication.